### PR TITLE
Adding Eddystone Support

### PR DESCRIPTION
**Pull request by BlackstoneEngineering**
This is the result of my work with Google to enable the eddystone beacon service and the eddystone config service.

### CODE REVIEW

File: ble/services/EddystoneConfigService.h [hunk 1]
```diff
@@ -0,0 +1,546 @@
...
+static const uint8_t UUID_LOCK_STATE_CHAR[]       = UUID_URI_BEACON(0x20, 0x81);
+static const uint8_t UUID_LOCK_CHAR[]             = UUID_URI_BEACON(0x20, 0x82);
+static const uint8_t UUID_UNLOCK_CHAR[]           = UUID_URI_BEACON(0x20, 0x83);
+static const uint8_t UUID_URI_DATA_CHAR[]         = UUID_URI_BEACON(0x20, 0x84);
+static const uint8_t UUID_FLAGS_CHAR[]            = UUID_URI_BEACON(0x20, 0x85);
+static const uint8_t UUID_ADV_POWER_LEVELS_CHAR[] = UUID_URI_BEACON(0x20, 0x86);
+static const uint8_t UUID_TX_POWER_MODE_CHAR[]    = UUID_URI_BEACON(0x20, 0x87);
+static const uint8_t UUID_BEACON_PERIOD_CHAR[]    = UUID_URI_BEACON(0x20, 0x88);
+static const uint8_t UUID_RESET_CHAR[]            = UUID_URI_BEACON(0x20, 0x89);
+extern const uint8_t       BEACON_EDDYSTONE[2];
```
Comment (rgrover) [inline ble/services/EddystoneConfigService.h]: can you align this declaration with the ones above?

File: ble/services/EddystoneConfigService.h [hunk 2]
```diff
@@ -0,0 +1,546 @@
...
+{
+public:    
+    /**
+     * @brief Transmission Power Modes for UriBeacon
+     */
+    static const uint8_t TX_POWER_MODE_LOWEST = 0; /*!< Lowest TX power mode */
+    static const uint8_t TX_POWER_MODE_LOW    = 1; /*!< Low TX power mode */
+    static const uint8_t TX_POWER_MODE_MEDIUM = 2; /*!< Medium TX power mode */
+    static const uint8_t TX_POWER_MODE_HIGH   = 3; /*!< High TX power mode */
+    static const unsigned NUM_POWER_MODES     = 4; /*!< Number of Power Modes defined */
```
Comment (rgrover) [inline ble/services/EddystoneConfigService.h]: how about an `enum {TX_POWER_MODE_LOWEST, TX_POWER_MODE_LOW, ..., NUM_POWER_MODES};`?

File: ble/services/EddystoneConfigService.h [hunk 3]
```diff
@@ -0,0 +1,546 @@
...
+
+    static const int ADVERTISING_INTERVAL_MSEC = 1000;  // Advertising interval for config service.
+    static const int SERVICE_DATA_MAX = 31;             // Maximum size of service data in ADV packets
+
+    typedef uint8_t Lock_t[16];               /* 128 bits */
+    typedef int8_t PowerLevels_t[NUM_POWER_MODES];
+
+    // There are currently 3 subframes defined, URI, UID, and TLM
+#define EDDYSTONE_MAX_FRAMETYPE 3
+    void (*frames[EDDYSTONE_MAX_FRAMETYPE])(uint8_t *, uint32_t);
```
Comment (rgrover) [inline ble/services/EddystoneConfigService.h]: should the following two members be public?

File: ble/services/EddystoneConfigService.h [hunk 4]
```diff
@@ -0,0 +1,546 @@
...
+    uint8_t frameIndex;
+    static const int URI_DATA_MAX = 18;
```
Comment (rgrover) [inline ble/services/EddystoneConfigService.h]: should the previous two members be public?

**Comment by BlackstoneEngineering** [comment]
Make whatever cosmetic changes you want to.

File: ble/services/EddystoneConfigService.h [hunk 5]
```diff
@@ -0,0 +1,546 @@
...
+        uint8_t       lockedState;
+        Lock_t        lock;
+        uint8_t       flags;
+        PowerLevels_t advPowerLevels; // Current value of AdvertisedPowerLevels
+        uint8_t       txPowerMode;    // Firmware power levels used with setTxPower()
+        uint16_t      beaconPeriod;
+        // TLM Frame Data
+        uint8_t       tlmVersion;     // version of TLM packet
+        bool          tlmEnabled;
+        float       tlmBeaconPeriod; // how often to broadcat TLM frame, in S
```
Comment (rgrover) [inline ble/services/EddystoneConfigService.h]: alignment

File: ble/services/EddystoneConfigService.h [hunk 6]
```diff
@@ -0,0 +1,546 @@
...
+        // URI Frame Data
+        uint8_t       uriDataLength;
+        UriData_t     uriData;
+        bool          uriEnabled;
+        float       uriBeaconPeriod; // how often to broadcast URIFrame, in S
```
Comment (rgrover) [inline ble/services/EddystoneConfigService.h]: alignment

File: ble/services/EddystoneConfigService.h [hunk 7]
```diff
@@ -0,0 +1,546 @@
...
+        // UID Frame Data
+        UIDNamespaceID_t uidNamespaceID; // UUID type, Namespace ID, 10B
+        UIDInstanceID_t  uidInstanceID;  // UUID type, Instance ID,  6B
+        bool          uidEnabled;
```
Comment (rgrover) [inline ble/services/EddystoneConfigService.h]: alignment

**Comment by rgrover** [comment]
OK. I'll make cosmetic changes. Could you please test the result? thanks.

**Pull request merged**